mermaid_py/client.py

```diff
@@ -209,118 +199,164 @@ def get_choices(self):
+        """
+        Gets the projects resource. Without query parameters, returns a list of projects of which
         the user is a member. The showall query parameter may be used to return projects unfiltered by the user’s
         membership. showall is important when the user is unauthenticated.
         :param showall: (optional) Returns all projects unfiltered by the user’s membership.
         :type showall: bool
         :return: projects.
         :rtype: dict
         """
-        if not showall:
-            return self._fetch_resource("projects")
-        else:
+        if showall:
             payload = "showall"
             return self._fetch_resource("projects", parameters=payload)
+        else:
+            return self._fetch_resource("projects")
 
-    def get_my_project(self, name=None, id=None):
+    def get_project(self, id: str = None, name: str = None):
         """
         Gets a specific MERMAID project. Either name or id required.
         See https://mermaid-api.readthedocs.io/en/latest/projects.html for more information on project resources.
+        :param id: (optional if 'name' provided) MERMAID project ID. If both id and name are provided, then id is used.
+        :type id: str
         :param name: (optional if 'id' provided) MERMAID project name.
         :type name: str
-        :param id: (optional if 'name' provided) MERMAID project ID.
-        :type id: str
         :return: project with associated data.
         :rtype: dict
         """
         if id:
-            return self._api_projects_path(id=id)
+            self._fetch_project_resource(id=id)
         elif name:
             projects = self.get_projects(showall=True)
-            proj = get_dict_by_keyval("name", name, projects["results"])
-            if proj is None:
-                raise Exception(f"No project with the name:{name}")
+            project = get_dict_by_keyval("name", name, projects["results"])
+            if project:
+                return project
             else:
-                return proj
+                raise InvalidProjectException(name=name)
 
-    def get_project_id(self, name):
+    def get_project_id(self, name: str):
         """
         Gets an ID for given MERMAID project.
         :param name: (optional if 'id' provided) MERMAID project name.
         :type: str
         :return: project ID.
         :rtype: str
         """
-        project = self.get_my_project(name)
-        # Nested if instead of 'elif name and self.get_my_project(name=name)' to improve performance by reducing
-        # the number of API calls for name or id NoneType check.
+        project = self.get_project(name=name)
+
         if project:
             return project.get("id")
+        else:
+            raise InvalidProjectException(name=name)
 
-    def get_project_data(self, data, name=None, id=None):
+    def get_project_resource(self, resource: str, id: str = None, name: str = None):
```

Comment:
I may be missing something, but I'm not seeing why you couldn't move the logic in _fetch_project_resource here and then get rid of _fetch_project_resource, and have everything else call this?